### ISSUE TITLE
Cannot proxy artifactory npm repository

### ISSUE THREAD

**Issue by VirtualTim** [sonatype/nexus-public#313]
I am running a Nexus 3 instance, and trying to proxy a remote, private, repository hosted using artifactory.
This repository requires authentication for all downloads. In fact, it will not even advertise which packages are available without authentication.

In npm, this is roughly my workflow to download the package @private/test:
```
npm config set @private:registry=https://my-nexus-instance/nexus/repository/npm-private/
npm install @private/test
```
npm then returns:
```
Not Found - GET https://my-nexus-instance/nexus/repository/npm-private/@private%2ftest - Package '@private/test' not found
```
(Note that setting the registry directly to the artifactory repository the `npm install` works)

Looking at the nexus logs I see this:
```
2024-01-11 10:24:07,450+0800 INFO  [qtp888189588-23429]  *UNKNOWN org.sonatype.nexus.repository.httpclient.internal.HttpClientFacetImpl - Repository status for npm-proxy-private changed from READY to AVAILABLE - reason n/a for n/a
2024-01-11 10:24:07,455+0800 WARN  [qtp888189588-23429]  *UNKNOWN com.sonatype.nexus.repository.npm.internal.NpmHandlers - Error: GET /@private/test: Status{successful=false, code=404, message='null'} - Invalid JSON input
[stacktrace]
Caused by: com.fasterxml.jackson.core.JsonParseException: Unexpected character ('<' (code 60)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
```

I notice than when attempting to browse the remote directly via a url like `https://artifactory.private.com/artifactory/product_release_npm/` I get redirected to `https://artifactory.private.com/ui/native/product_release_npm/`.

Googling similar issues direct me to https://issues.sonatype.org/browse/NEXUS-31268 and https://issues.sonatype.org/browse/NEXUS-8745, but those appear to have now been deleted.

**Comment by VirtualTim** [comment]
I suspect that the `Unexpected character ('<' (code 60))` message means that the get request is returning html. Is there a way to view what actually gets returned?

**Comment by VirtualTim** [comment]
Changing the "Remote Storage" URL from `https://artifactory.private.com/artifactory/product_release_npm/` to `https://artifactory.private.com/artifactory/api/npm/product_release_npm` results in this message in the logs:
```
[11/Jan/2024:11:31:02 +0800] myUsername "GET https://artifactory.private.com/artifactory/api/npm/product_release_npm/@private%2ftest HTTP/1.1" 403 -1 309 "Nexus/3.61.0-02 (OSS; Windows Server 2019; 10.0; amd64; 1.8.0_332) " [qtp888189588-22853]
```

However when I run:
```
curl -u myUsername:password https://artifactory.private.com/artifactory/api/npm/product_release_npm/@private%2ftest
```
That succeeds.

**Comment by VirtualTim** [comment]
I got this working by using
`Authentication Type` = `Preemptive Bearer Token`, and
`Remote Storage` = `https://artifactory.private.com/artifactory/api/npm/product_release_npm`. (note the `api`).

I don't know why `Authentication Type` = `Username` did not work.

Hopefully this helps someone else in the future.

**Comment by anatoliy-v** [comment]
"Authentication Type = Username" 
Doesn't work for me too and I can't use "Preemptive Bearer Token" for Google Artifact Registry(
According to the log - Nexus ignores "Authentication Type = Username" completely.
I've disabled authentication and add custom header with my json key to the field User-agent customization 
and it works like a magic. 
`-u _json_key_base64:****token****`
I suggest to open this ticket again and fix "Authentication Type = Username"
My Nexus version - OSS 3.41.1-01

**Comment by igor-sweet** [comment]
We found the same problem with Helm repositories and NPM repositories. 
While debugging the request with an nginx proxy, we could see that no basic authentication credentials were being sent for these requests. 

`1.2.3.4 - - [11/Jun/2024:11:57:58 +0200] "GET /artifactory/api/npm/`

We did not try to change the authentication method to "Peemptive Bearer Token" in NPM, but in Helm proxy repos there is only username or Windows NTLM.

It's really misleading and strange that the Nexus log shows the request with user information.
